Spigot-Server-Patches/0496-Allow-delegation-to-vanilla-chunk-gen.patch

```diff
@@ -0,0 +1,71 @@
+From 9a92b34331e5f08498239aa29b6bba768fdba252 Mon Sep 17 00:00:00 2001
+From: MiniDigger <admin@minidigger.me>
+Date: Wed, 29 Apr 2020 02:10:32 +0200
+Subject: [PATCH] Allow delegation to vanilla chunk gen
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index f49193d9d..c7e41c7e3 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -1857,6 +1857,31 @@ public final class CraftServer implements Server {
+         return new CraftChunkData(world);
+     }
+ 
++    // Paper start
++    @Override
++    public ChunkGenerator.ChunkData createVanillaChunkData(World world, int x, int z) {
```

Comment:
is there anything here making this unsafe for async usage? If so, i think we should resolve that.

WorldGenerators are now supposed to be parallel capable, so let's not regress that.

Reply:
Overall may be better to return a CompletableFuture for safety if we do need to multi tier stage this int he future.

Reply:
Even if that future will currently be resolved immediately, keeps API safe if it can't be in future.

Reply:
We really do NEED the ability to get the content from the vanilla generator as modified by "shouldAddFoo". Otherwise generators still need to come up with that content on their own anyways.

As for the async, iirc the vanilla generator has the isParallelEnabledOrWhateverItWasCalled function returning false. I assume this would be doing the same here.

Reply:
![image](https://user-images.githubusercontent.com/1046026/80663420-c7a55100-8a48-11ea-93e6-c8c0ac38d9e7.png)
I'll start devving like this and let you know if it breaks...

Reply:
I don't know enough about world generation and how bukkits custom world gen fits into the whole parallel world gen. I did just assume this will work as it's what vanilla does too, altho in multiple steps after each other, this now does all steps per chunk.

In my testing I didn't run into any issues (as long as I left decorations alone, those blow up horrible, but spigot has the booleans for that you can override), pangammas testing is much more large scale, so let's see what that does.

If anybody who has more insight into how this all works together that would be greatly appreciated obviously.

Reply:
It does not break, though it is always possible I am just not testing hard enough.

Reply:
I have been testing with parallel capable enabled for the last 3 days or so. I have not experienced any crashes from leaving that flag true.